File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> map = new HashMap<>();
        String key;
        List<String> list;
        char[] arr;
        for(String str : strs) {
            arr = str.toCharArray();
            Arrays.sort(arr);
            key = new String(arr);
            list = map.getOrDefault(key, new ArrayList<>());
            list.add(str);
            map.put(key, list);
        }
        return new ArrayList<List<String>>(map.values());
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        def hashWord(word):
            ret = 0
            sortedWord = sorted(word)
            for i in range(len(sortedWord)):
                ret += ord(sortedWord[i]) * (26**i)
            return ret
        
        wordDict = {}
        for word in strs:
            ind = hashWord(word)
            value = wordDict.get(ind, [])
            value.append(word)
            wordDict[ind] = value
        wordSet = []
        for ind, key in enumerate(wordDict):
            wordSet.append(wordDict[key])
        return wordSet

